lambdas/service/app.py

```diff
@@ -315,6 +315,40 @@ def validate_params(query_params: Mapping[str, str],
                 raise BadRequestError(f'Invalid input type for `{param_name}`')
 
 
+@app.route('/integrations', methods=['GET'], cors=True)
+def get_integrations():
+    query_params = app.current_request.query_params or {}
+    validate_params(query_params, entity_type=str, integration_type=str)
+    try:
+        entity_type = query_params['entity_type']
+        integration_type = query_params['integration_type']
+    except KeyError:
+        raise BadRequestError('Parameters entity_type and integration_type must be given')
+    body = _fetch_integrations(entity_type, integration_type)
+    return Response(status_code=200,
+                    headers={"content-type": "application/json"},
+                    body=json.dumps(body))
+
+
+def _fetch_integrations(entity_type, integration_type):
+    def set_entity_ids(integration):
+        if 'entity_ids' in integration:
+            integration['entity_ids'] = integration['entity_ids'][config.dss_deployment_stage]
+        return integration
```

Comment:
You should not modify the argument and return it. That's redundant and confusing. If you want to use this function with `map` it should be [pure](https://en.wikipedia.org/wiki/Pure_function).

Reply:
Re-wrote section to grab copies of the relevant data from `portals_db` instead of references which required modifying the source data